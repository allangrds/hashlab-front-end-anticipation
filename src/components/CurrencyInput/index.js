import React from 'react'
import PropTypes from 'prop-types'
import CurrencyInput from 'react-currency-input'
import classNames from 'classnames'

import styles from './styles.module.css'

const Input = ({
  errorMessage,
  label,
  required,
  tip,
}) => {
  const bottomText = errorMessage || tip
  const bottomClassName = errorMessage
    ? styles.error
    : styles.tip
  const inputClassName = errorMessage
    ? classNames(styles.input, styles['input-error'])
    : styles.input

  return (
    <fieldset className={styles.fieldset}>
      <label
        className={styles.label}
        htmlFor="Aname"
      >
        { label } { required && '*' }
      </label>
      <CurrencyInput
        decimalSeparator=","
        thousandSeparator="."
        prefix="R$"
        id="Aname"
        className={inputClassName}
      />
      <p className={bottomClassName}>
        { bottomText }
      </p>
    </fieldset>
  )
}

Input.propTypes = {
  errorMessage: PropTypes.string,
  label: PropTypes.string.isRequired,
  required: PropTypes.bool,
  tip: PropTypes.string,
}

Input.defaultProps = {
  errorMessage: null,
  required: false,
  tip: null,
}

export default Input
