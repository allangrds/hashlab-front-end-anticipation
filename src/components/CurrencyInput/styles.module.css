@import "../../assets/settings/colors.css";

.fieldset {
  border: none;
}

.label {
  font-family: "Source Sans Pro", sans-serif;
  color: var(--color-gray-1);
  font-size: 14px;
  margin-bottom: 6px;
  display: block;
}

.input {
  font-family: "Source Sans Pro", sans-serif;
  color: var(--color-black);
  font-size: 14px;
  padding: 10px 0 9px 14px;
  border: 1px solid var(--color-gray-4);
  border-radius: 4px;
  outline: none;
  width: 100%;
}

.input:focus {
  border-color: var(--color-blue-3);
}

.input-error {
  border-color: var(--color-error);
}

.tip {
  color: var(--color-gray-2);
  font-family: "Source Sans Pro", sans-serif;
  font-size: 11px;
  margin-top: 3px;
  margin-bottom: 0;
}

.error {
  color: var(--color-error);
  font-family: "Source Sans Pro", sans-serif;
  font-size: 11px;
  margin-top: 3px;
  margin-bottom: 0;
}
