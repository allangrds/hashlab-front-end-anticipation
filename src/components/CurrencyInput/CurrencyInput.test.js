import React from 'react'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import CurrencyInput from '.'

test('CurrencyInput should render and be updated correctly', () => {
  const props = {
    label: 'Esse é o label',
    required: true,
    tip: 'Máximo de 12 parcelas',
  }
  const initialValue = 'R$0,00'
  const newRawValue = '123'
  const newFormatedValue = 'R$1,23'

  const {
    container,
    getByDisplayValue,
    getByLabelText,
    getByText,
  } = render(
    <CurrencyInput {...props} />,
  )

  getByLabelText(`${props.label} *`)
  getByText(props.tip)

  const input = getByDisplayValue(initialValue)

  userEvent.type(input, newRawValue)

  getByDisplayValue(newFormatedValue)

  expect(container).toMatchSnapshot()
})

test('CurrencyInput should render with error message', () => {
  const props = {
    errorMessage: 'Mensagem de erro',
    label: 'Esse é o label',
    required: true,
  }

  const {
    container,
    getByText,
  } = render(
    <CurrencyInput {...props} />,
  )

  getByText(props.errorMessage)

  expect(container).toMatchSnapshot()
})
