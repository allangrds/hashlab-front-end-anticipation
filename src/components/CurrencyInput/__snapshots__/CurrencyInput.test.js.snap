// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyInput should render and be updated correctly 1`] = `
<div>
  <fieldset
    class="fieldset"
  >
    <label
      class="label"
      for="Aname"
    >
      Esse é o label
       
      *
    </label>
    <input
      class="input"
      id="Aname"
      type="text"
      value="R$1,23"
    />
    <p
      class="tip"
    >
      Máximo de 12 parcelas
    </p>
  </fieldset>
</div>
`;

exports[`CurrencyInput should render with error message 1`] = `
<div>
  <fieldset
    class="fieldset"
  >
    <label
      class="label"
      for="Aname"
    >
      Esse é o label
       
      *
    </label>
    <input
      class="input input-error"
      id="Aname"
      type="text"
      value="R$0,00"
    />
    <p
      class="error"
    >
      Mensagem de erro
    </p>
  </fieldset>
</div>
`;
